public/app/core/components/DeleteButton/DeleteButton.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { Component } from 'react';
+
+export default class DeleteButton extends Component<any, any> {
+  state = {
+    deleteButton: 'delete-button--show',
+    confirmSpan: 'confirm-delete--removed',
+  };
+
+  handleDelete = event => {
+    if (event) {
+      event.preventDefault();
+    }
+
+    this.setState({
+      deleteButton: 'delete-button--hide',
+    });
+
+    setTimeout(() => {
+      this.setState({
```

Comment:
You could probably remove the `--removed` property, and always display the confirm button invisibly and simply have a `onClickConfirm` handler that checks if `showConfirm` is true, and only then execute the `confirmDelete` prop.